kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java

```diff
@@ -183,6 +187,29 @@ private HitsMetadata getHits(KaldbSearch.SearchResult searchResult) throws IOExc
   @Get
   @Path("/:indexName/_mapping")
   public HttpResponse mapping(@Param("indexName") Optional<String> indexName) throws IOException {
+    // todo - support query params including dataset and time window - this will require
+    //  rework of the Grafana plugin and will deviate from the Opensearch api spec
+
+    // Use a tree map so the results are naturally sorted
+    Map<String, Map<String, String>> propertiesMap = new TreeMap<>();
+    KaldbSearch.SchemaResult schemaResult =
+        searcher.getSchema(
+            KaldbSearch.SchemaRequest.newBuilder()
+                .setDataset("*")
```

Comment:
Wouldn't this only work if data from the past hour has been ingested? If so, that would mean that clusters with only historical data (or in the middle of being bootstrapped) couldn't find schema information.

Also, are we assuming that schema is immutable over time?

Reply:
We do expect that the schema is mutable over time, and indeed there exists a possibility that two chunks may have different schema for a newly emerging field if it were being published in multiple formats (ie, `int` and `string`). The expectation is that you can use the KalDB API to query for a given time window, and get a schema that applies specifically to that time.

As of this PR, the GRPC api does allow you to query for a given time range and it return schema specific to that window. The Opensearch compatible API (and our fork of the Grafana Opensearch plugin) do not support passing date params, as they do not expect the schema to not change over time. As the Trino connector proof-of-concept is currently using this Opensearch API, I can certainly expose these as optional url params on this endpoint similar to the index name. I would expect we would want to extend our fork of the Grafana plugin to start passing these date params anyhow.

As to your other question regarding bootstrapping - I'm not certain this is necessarily a problem for KalDB. My expectation would be that the request for schema would follow what data is available for query - that is, if the data isn't loaded then the schema of the system would only reflect that which was.